website/src/components/DataUseTerms/EditDataUseTermsModal.tsx

```diff
@@ -0,0 +1,289 @@
+                        just restricted sequences. You can use the filters to do so.
+                    </p>
+                    <CancelSubmitButtons
+                        clientConfig={clientConfig}
+                        accessToken={accessToken}
+                        newTerms={{ type: openDataUseTermsType }}
+                        affectedAccesions={state.restrictedAccessions}
+                        closeDialog={closeDialog}
+                    />
+                </div>
+            );
+        case 'allRestricted':
+            const earliestDateDisplay = state.earliestRestrictedUntil!.toFormat('yyyy-MM-dd');
+            return (
+                <div className='space-y-4'>
+                    <ActiveFilters sequenceFilter={sequenceFilter} />
+                    <h4 className='font-bold mb-2'>
+                        Choose the new data use terms for {state.restrictedCount} restricted sequence
+                        {state.restrictedCount > 1 ? 's' : ''}
+                    </h4>
+                    <p className='text-xs text-gray-500'>
+                        The release date of a sequence cannot be updated to be later than the date that is currently
+                        set. This means that the new release date can only be between now and the earliest release date
+                        for any of the selected sequences, which is <b>{earliestDateDisplay}</b>.
+                    </p>
+                    <div className='flex flex-row'>
+                        <DataUseTermsSelector
+                            maxRestrictedUntil={state.earliestRestrictedUntil!}
+                            setDataUseTerms={setDataUseTerms}
+                        />
+                    </div>
+                    <CancelSubmitButtons
+                        clientConfig={clientConfig}
+                        accessToken={accessToken}
+                        newTerms={dataUseTerms}
+                        affectedAccesions={state.restrictedAccessions}
+                        closeDialog={closeDialog}
+                    />
+                </div>
+            );
+    }
+};
+
+interface CancelSubmitButtonProps {
+    clientConfig: ClientConfig;
+    accessToken: string;
+    newTerms: DataUseTerms | null;
+    affectedAccesions: string[];
+    closeDialog: () => void;
+}
+
+const CancelSubmitButtons: React.FC<CancelSubmitButtonProps> = ({
+    clientConfig,
+    accessToken,
+    closeDialog,
+    newTerms,
+    affectedAccesions,
+}) => {
+    const setDataUseTermsHook = backendClientHooks(clientConfig).useSetDataUseTerms(
+        { headers: createAuthorizationHeader(accessToken) },
+        {
+            onError: (error) =>
+                toast.error('Failed to edit terms of use: ' + stringifyMaybeAxiosError(error), {
+                    position: 'top-center',
+                    autoClose: false,
+                }),
+            onSuccess: () => location.reload(),
```

Comment:
Fairly important change: we need to make a toast for the success operation here or things are super unclear. Maybe for now as a kind of hack: 

```
onSuccess: () => {
toast.success("Data use terms updated successfully, page will refresh in 5 seconds");
setTimeout(()=>location.reload(),5000);
}
```

Reply:
I added a popup with info [`c716111` (#3322)](https://github.com/loculus-project/loculus/pull/3322/commits/c716111f5e7acde1c506350152a289f1a67f8b47)

I don't think we need the timeout for reload, I just removed the reload alltogether (not sure why we would reload the page. I got this from the EditDataUseTermsButton, but since the data won't be there, I think it doesn't make sense to reload)

Reply:
I have refactored both places to use the same toasts: [`9a56b19` (#3322)](https://github.com/loculus-project/loculus/pull/3322/commits/9a56b19443ebfb136f7557f9b9c98b3d99343161)

Reply:
Yeah, the reload did make sense in the context of an individual sequence details page , as the sequence details page is instaneously up to date. But when the sequence details are open in a modal it loses the modal, so that's bad anyway so I think your new approach sounds good for now.